autogluon_zeroshot/simulation/tabular_predictions.py

```diff
@@ -362,25 +710,24 @@ def rename_datasets(self, rename_dict: dict):
 
     @staticmethod
     def _save_metadata(output_dir, dataset_to_models):
-        with open(output_dir / "metadata.json", "w") as f:
-            metadata = {
-                "dataset_to_models": dataset_to_models,
-            }
-            json.dump(metadata, f)
+        metadata = {
+            "dataset_to_models": dataset_to_models,
+        }
+        save_pkl(path=str(output_dir / "metadata.pkl"), object=metadata)
```

Comment:
If there is no unit test or issue that we can pinpoint to, can we keep json? Pkl is not safe without precaution as opposed to json (also json is readable in clear).

Reply:
This is the problem: https://stackoverflow.com/questions/1450957/pythons-json-module-converts-int-dictionary-keys-to-strings

Reply:
Example of how this breaks our current code:

```python3
import json

dataset_to_models = {
    377590: {  # dataset TID
        0: ['model1', 'model2'],  # fold to models
        1: ['model1', 'model2'],  # fold to models
    }
}
json_dataset_to_models = json.dumps(dataset_to_models)

load_dataset_to_models = json.loads(json_dataset_to_models)

print(f'Original:')
print(dataset_to_models)
print(f'JSON Loaded:')
print(load_dataset_to_models)

assert dataset_to_models == load_dataset_to_models

```

Out:

```
Original:
{377590: {0: ['model1', 'model2'], 1: ['model1', 'model2']}}
JSON Loaded:
{'377590': {'0': ['model1', 'model2'], '1': ['model1', 'model2']}}
Traceback (most recent call last):
  File "/home/ubuntu/workspace/code/scratch/scratch1.py", line 18, in <module>
    assert dataset_to_models == load_dataset_to_models
AssertionError
```

Reply:
If you prefer to keep using JSON, we would need to either:

1. Figure out how to do it with integers kept in tact (seems not possible without a lot of hacking)
2. Convert all integer keys to strings (would prefer not doing this though)